examples: addition of Flask-Menu and DB commit

jacquerie opened the pull request:
While investigating another issue in another package I needed to have a minimal application with records, record views, and permissions. The two examples here were what I needed, but required some tweaks to be run, which I'm sharing back.

On a different note, it would be very appreciated if the `examples` folder contained a `requirements.txt` file that a developer can `pip install -r` in a new virtualenv and have all the dependencies required to run the examples.

jirikuncar commented on examples/app.py:
```diff
@@ -148,3 +147,5 @@ def records():
         PersistentIdentifier.create(
             'recid', '7', status=PIDStatus.RESERVED)
+
+        db.session.commit()
```
It should be probably outside the nested transaction.

jacquerie replied:
Whoops, it was, but then I changed it for no good reason.

jirikuncar commented on examples/permsapp.py:
```diff
@@ -56,6 +57,7 @@
 from __future__ import absolute_import, print_function
 
 from app import app, fixtures, rec1_uuid
+from flask_menu import Menu
```
why do you need the flask-menu?

jacquerie replied:
Without these two lines I get, when running `$ FLASK_APP=examples/permsapp.py flask run --debugger` and navigating to `/records/1`:

```python
[2016-07-15 13:28:59,415] ERROR in app: Exception on /records/1 [GET]
Traceback (most recent call last):
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/flask/app.py", line 1634, in full_dispatch_request
    self.try_trigger_before_first_request_functions()
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/flask/app.py", line 1660, in try_trigger_before_first_request_functions
    func()
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/invenio_accounts/views.py", line 67, in init_menu
    item = current_menu.submenu('breadcrumbs.settings')
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/werkzeug/local.py", line 343, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/werkzeug/local.py", line 302, in _get_current_object
    return self.__local()
  File "/Users/jacquerie/.virtualenvs/access/lib/python2.7/site-packages/flask_menu/__init__.py", line 65, in root
    return current_app.extensions['menu']
KeyError: 'menu'
127.0.0.1 - - [15/Jul/2016 13:28:59] "GET /records/1 HTTP/1.1" 500 -
```

jirikuncar commented on examples/permsapp.py:
```diff
@@ -103,4 +106,5 @@ def access():
     db.session.add(ActionUsers(
         action=records_read_all.value, argument=rec1_uuid,
         user=reader, exclude=True))
+
```
?

jirikuncar commented on examples/app.py:
```diff
@@ -30,10 +30,11 @@
-   $ flask -a app.py db create
-   $ flask -a app.py fixtures records
-   $ flask -a app.py --debug run
+   $ EXPORT FLASK_APP=app.py
```
`s/EXPOXT/export/`

jacquerie commented:
Following an [informal RFC](https://gitter.im/inveniosoftware/invenio?at=578c2b2d196179690efc8cc9) I also added a commit with a `requirements.txt` obtained via `pip freeze`. Added here because it assumes the other documentation updates to be there.

jacquerie commented on examples/requirements.txt:
```diff
@@ -0,0 +1,54 @@
+amqp==1.4.9
```
TODO because of https://github.com/inveniosoftware/invenio-records-rest/pull/103#discussion_r71332337

jacquerie replied:
`requirements.txt` has been dried up.

jirikuncar commented:
@jacquerie please check `MANIFEST.in` (`examples/requirements.txt`).

jacquerie commented:
>  @jacquerie please check MANIFEST.in (examples/requirements.txt).

Whoops. Sorry for not running the tests locally, previously Travis was failing for unrelated reasons, so I thought the test suite was broken.
